tests/framework/results/test_manager.py

```diff
@@ -364,6 +362,10 @@ def _check_house_points(pop: pd.DataFrame, step_number: int) -> None:
         """We know that house points are stratified by 'student_house' and 'power_level'.
         and that each wizard of gryffindor and of level 50 and 80 gains a point
         """
+        # Need the stratification columns to be Categorical
```

Comment:
Why do they need to be categorical? I don't see any obvious reason for it looking at the code here.

Reply:
Because the very last assertion checks that the dataframes are equal and they're not if they don't have the same multiindex dtypes

Reply:
Why can't we just assert that all the values are equal?